tests/test_google_ads_interrupted_sync_remove_stream.py

```diff
@@ -0,0 +1,264 @@
+
+        Expected State Structure:
+            state = {'currently_syncing': ('<stream-name2>', '<customer-id>'),
+            'bookmarks': {
+                '<stream-name1>': {'<customer-id>': {'<replication-key>': <completed-bookmark-value>}},
+                '<stream-name2>': {'<customer-id>': {'<replication-key>': <incomplete-bookmark-value>}},
+
+        Test Cases:
+         - Verify behavior is consistent when a stream is removed from selected list between initial and resuming sync
+         - Verify an interrupted sync can resume based on the `currently_syncing` and stream level bookmark value
+         - Verify only records with replication-key values greater than or equal to the stream level bookmark are replicated on the resuming sync for the interrupted stream
+         - Verify the yet-to-be-synced streams are replicated following the interrupted stream in the resuming sync. All yet-to-be-synced streams must replicate before streams that were already synced.
+        """
+        print("Interrupted Sync Test for tap-google-ads with added stream")
+
+        # the following streams are under test as they all have 4 consecutive days with records e.g.
+        # ('2022-01-23T00:00:00.000000Z', '2022-01-23T00:00:00.000000Z', '2022-01-24T00:00:00.000000Z', '2022-01-25T00:00:00.000000Z')])}
+        streams_under_test = {'account_performance_report',
+                              'search_query_performance_report',
+                              'user_location_performance_report',
+        }
+
+        # Create connection using a recent start date
+        self.start_date = '2022-01-22T00:00:00Z'
+        conn_id = connections.ensure_connection(self, original_properties=False)
+
+        # Run a discovery job
+        found_catalogs_1 = self.run_and_verify_check_mode(conn_id)
+
+        # partition catalogs for use in table/field seelction
+        test_catalogs_1 = [catalog for catalog in found_catalogs_1
+                           if catalog.get('stream_name') in streams_under_test]
+        core_catalogs_1 = [catalog for catalog in test_catalogs_1
+                           if not self.is_report(catalog['stream_name'])]
+        report_catalogs_1 = [catalog for catalog in test_catalogs_1
+                             if self.is_report(catalog['stream_name'])]
+
+        # select all fields for core streams
+        self.select_all_streams_and_fields(conn_id, core_catalogs_1, select_all_fields=True)
+
+        # select 'default' fields for report streams
+        self.select_all_streams_and_default_fields(conn_id, report_catalogs_1)
+
+        # Run a sync
+        full_sync = self.run_and_verify_sync(conn_id)
+
+        # acquire records from target output
+        full_sync_records = runner.get_records_from_target_output()
+        full_sync_state = menagerie.get_state(conn_id)
+
+        # Add a stream between syncs
+        removed_stream = 'user_location_performance_report'
```

Comment:
I think we should use the stream that was synced rather than the one that was not. Then we can verify the state saves without that stream on the resuming sync.

Reply:
Agreed, updated.